zcash_primitives/src/transaction/builder.rs

```diff
@@ -68,6 +69,15 @@ pub enum Error<FeeError> {
     TransparentBuild(transparent::builder::Error),
     /// An error occurred in constructing the Sapling parts of a transaction.
     SaplingBuild(sapling::builder::Error),
+    /// An error occurred in constructing the Orchard parts of a transaction.
+    OrchardBuild(orchard::builder::Error),
+    /// An error occurred in adding an Orchard Spend a transaction.
+    OrchardSpend(&'static str),
+    /// An error occurred in adding an Orchard Output a transaction.
+    OrchardRecipient(&'static str),
```

Comment:
Using strings to communicate these errors does not permit programmatic analysis without parsing the strings; it would be better to add these cases to `orchard::builder::Error` or to add a separate enum that carries appropriate well-typed data describing the errors.

Reply:
https://github.com/zcash/orchard/pull/366 adds concrete error types here. This should be the last of the simple changes to orchard.

Reply:
Hi @nuttycom I added a review to [orchard #366](https://github.com/zcash/orchard/pull/366).  Let me know if there's something else I can do to move that work along!  Thanks.